src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests_ListPatterns.cs

```diff
@@ -8322,4 +8346,217 @@ public void ListPattern_NullTestOnSlice()
             Diagnostic(ErrorCode.ERR_SwitchCaseSubsumed, "[1,2,3]").WithLocation(15, 10)
             );
     }
+
+    [Fact, WorkItem(59465, "https://github.com/dotnet/roslyn/issues/59465")]
+    public void MissingLength()
+    {
+        var source = @"
+_ = new S() is [];
+_ = new S() is [..];
+_ = new S() is [0, .. var x, 1];
+
+struct S
+{
+    public int this[System.Index i] => 0;
+    public int this[System.Range i] => 0;
+}
+";
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics(
+            // (2,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[]").WithArguments("S").WithLocation(2, 16),
+            // (3,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [..];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[..]").WithArguments("S").WithLocation(3, 16),
+            // (4,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [0, .. var x, 1];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[0, .. var x, 1]").WithArguments("S").WithLocation(4, 16)
+            );
+    }
+
+    [Fact, WorkItem(59465, "https://github.com/dotnet/roslyn/issues/59465")]
+    public void MissingLength_WithIntIndexer()
+    {
+        var source = @"
+_ = new S() is [];
+
+struct S
+{
+    public int this[int i] => i;
+}
+";
+        var comp = CreateCompilationWithIndexAndRangeAndSpan(source);
+        comp.VerifyEmitDiagnostics(
+            // (2,16): error CS8985: List patterns may not be used for a value of type 'S'. No suitable 'Length' or 'Count' property was found.
+            // _ = new S() is [];
+            Diagnostic(ErrorCode.ERR_ListPatternRequiresLength, "[]").WithArguments("S").WithLocation(2, 16),
+            // (2,16): error CS1503: Argument 1: cannot convert from 'System.Index' to 'int'
+            // _ = new S() is [];
+            Diagnostic(ErrorCode.ERR_BadArgType, "[]").WithArguments("1", "System.Index", "int").WithLocation(2, 16)
+            );
+    }
+
+    [Fact, WorkItem(59465, "https://github.com/dotnet/roslyn/issues/59465")]
+    public void MissingLength_PrivateLength()
```

Comment:
>MissingLength_PrivateLength [](http://example.com/codeflow?start=16&length=27)

It looks like all tests below, but the last one, target ```System.Index``` indexer. Do we have similar tests for ```int``` indexer?

Reply:
Yes, all the previous tests for those scenarios (bad Length, such as `ListPattern_StringLength`, `SlicePattern_VoidReturn`, `SlicePattern_LengthAndIndexAndSliceAreStatic`, `ListPattern_UintCount`, `ListPattern_LengthFieldNotApplicable`, ...) used `int` Indexer. That's why this problem was missed.